components/gateway/cmd/main.go

```diff
@@ -105,16 +105,14 @@ func initAuditLogsSvc(done chan bool) AuditogService {
 	err := envconfig.InitWithPrefix(&auditlogCfg, "APP")
 	exitOnError(err, "Error while loading auditlog cfg")
 
-	httpClient, tenant := initAuditlogClient(auditlogCfg)
-	uuidSvc := uuid.NewService()
-	timeSvc := &time.TimeService{}
+	httpClient, msgFactory := initAuditlogClient(auditlogCfg)
```

Comment:
```
	httpClient, msgFactory := initAuditlogClient(auditlogCfg)

	auditlogClient, err := auditlog.NewClient(auditlogCfg, httpClient)
```
don't you think that it looks strange? You initAuditLog, and then create `audiitlogClient`